src/odemis/acq/stream/_base.py

```diff
@@ -1302,7 +1302,8 @@ def getPixelCoordinates(self, p_pos):
         tform = AffineTransform(matrix, translation)
         pixel_pos_c = tform.inverse().apply(p_pos)
         # a "-" is used for the y coordinate because Y axis has the opposite direction in physical coordinates
-        pixel_pos = int(pixel_pos_c[0] + size[0] / 2), - int(pixel_pos_c[1] - size[1] / 2)
+        pixel_pos = (int(math.floor(pixel_pos_c[0] + size[0] / 2)),
+                     int(math.floor(- (pixel_pos_c[1] - size[1] / 2))))
```

Comment:
Now that we are touching this code anyway, maybe explain with a comment why we are adding `shape / 2` in the mix here. I can imagine it is because we want the origin to be in the image center, but a comment could really help clarifying intentions.

Reply:
I added a comment. Indeed, it's because the image metadata refers to the center of the image, while the pixel coordinates are always relative to the top-left pixel.